ISSUE-22 Add bindings to `dkim_libfeature()`

leroycep opened the pull request:
Solves #22, mostly. Adds the function `lib_feature()` and a test. Documentation still needs to be added.

- [x] code
- [x] tests
- [x] documentation

leroycep committed:
ISSUE-22 Add bindings to `dkim_libfeature()`

godsflaw commented on index.js:
```diff
@@ -50,4 +50,8 @@ OpenDKIM.prototype.tmpdir = function (data) {
 };
 
+OpenDKIM.prototype.lib_feature = function (feature) {
+  this.lib_feature_obj({feature: feature});
```
I might end up doing this `*_obj` for other native methods too.  I'm looking into making a bunch of the interface async with Promises.

godsflaw commented on test/00-administration/02-libfeature.js:
```diff
@@ -0,0 +1,21 @@
+    opendkim.lib_feature('DKIM_FEATURE_QUERY_CACHE');
+    opendkim.lib_feature('DKIM_FEATURE_SHA256');
+    opendkim.lib_feature('DKIM_FEATURE_DNSSEC');
+    opendkim.lib_feature('DKIM_FEATURE_OVERSIGN');
```
Did you end up validating these outputs?  I know the results would be non-deterministic to test against, as this just tells you what libopendkim was compiled with.  I just want to make sure they tossed back what you expected?

godsflaw replied:
hmm, if I wrap those in a `console.log()` I get a series of `undefined` values:
```
    console.log(opendkim.lib_feature('DKIM_FEATURE_DIFFHEADERS'));              
    console.log(opendkim.lib_feature('DKIM_FEATURE_PARSE_TIME'));               
    console.log(opendkim.lib_feature('DKIM_FEATURE_QUERY_CACHE'));              
    console.log(opendkim.lib_feature('DKIM_FEATURE_SHA256'));                   
    console.log(opendkim.lib_feature('DKIM_FEATURE_DNSSEC'));                   
    console.log(opendkim.lib_feature('DKIM_FEATURE_OVERSIGN'));                 
```
```
undefined
undefined
undefined
undefined
undefined
undefined
  ✔ test lib feature

  1 test passed [23:13:17]
```

godsflaw replied:
to catch this better, we should make sure all of these checks test for type boolean
```
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_DIFFHEADERS'), 'boolean');                            
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_PARSE_TIME'), 'boolean');    
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_QUERY_CACHE'), 'boolean');   
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_SHA256'), 'boolean');        
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_DNSSEC'), 'boolean');        
    t.is(typeof opendkim.lib_feature('DKIM_FEATURE_OVERSIGN'), 'boolean');
```

godsflaw reviewed (commented):
Not sure how to handle the documentation on this one.    Since it's just markdown, you could toss it in a gist and I'll copy it into the wiki.

godsflaw commented:
I have to figure out what's up with CI on this one, I figure this is because the PR is from your fork and so it has not figured out it has to run.

leroycep committed:
Make `lib_feature` return a value

I kept editing `lib_feature_obj()`, but it turned out that I forgot the return
statement in the wrapper `lib_feature()`.

leroycep commented:
Okay, I updated the tests and fixed the code. Turns out, I forgot the return statement in the `lib_feature()` wrapper function. :P

Now to write the docs.

leroycep commented:
Wrote some docs! https://gist.github.com/Geemili/7ba3f0c430b9cdc5120a62533271c3bb

Would add a pull request for it, but GitHub doesn't seem to support that. :(

godsflaw reviewed (approved):
+1

godsflaw commented:
Added the wiki page for this PR:
https://github.com/godsflaw/node-opendkim/wiki/opendkim.lib_feature()
